File: 프로그래머스/1/12917. 문자열 내림차순으로 배치하기/문자열 내림차순으로 배치하기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        
        char[] c = s.toCharArray();
        int len = c.length;
        
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < len - 1; i++){
            for(int j = 0; j < len - i - 1; j++){
                if(c[j] < c[j + 1]){
                    char temp = c[j];
                    c[j] = c[j + 1];
                    c[j + 1] = temp;
                }
            }
        }
        
        for(char cc : c){
            sb.append(cc);
        }
        
        answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/1/12917. 문자열 내림차순으로 배치하기/문자열 내림차순으로 배치하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(const char* s) {
    int len = strlen(s);
    char* answer = (char*)malloc((len + 1) * sizeof(char));
    strcpy(answer, s);
    
    for(int i = 0; i < len - 1; i++){
        for(int j = i + 1; j < len; j++){
            if(answer[i] < answer[j]){
                int temp = answer[i];
                answer[i] = answer[j];
                answer[j] = temp;
            }
        }
    }
    
    return answer;
}